kroxylicious-filters/kroxylicious-record-encryption/src/main/java/io/kroxylicious/filter/encryption/crypto/Wrapper.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.filter.encryption.crypto;
+
+import java.nio.ByteBuffer;
+import java.util.Set;
+import java.util.function.BiConsumer;
+import java.util.function.BiFunction;
+
+import org.apache.kafka.common.header.Header;
+import org.apache.kafka.common.record.Record;
+import org.apache.kafka.common.record.RecordBatch;
+
+import io.kroxylicious.filter.encryption.config.ParcelVersion;
+import io.kroxylicious.filter.encryption.config.RecordField;
+import io.kroxylicious.filter.encryption.dek.CipherManager;
+import io.kroxylicious.filter.encryption.dek.Dek;
+import io.kroxylicious.kms.service.Serde;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+public interface Wrapper {
```

Comment:
it'd be good to get java doc on this interface

Reply:
Absolutely. However, I'm still working out some of the details around how we separate the specification of these things from their implementation. With luck I might have something working today.

Reply:
@k-wall I've pushed some abstractions (`PersistedIdentifiable`, `Resolver`, `AbstractResolver`) which I think is along the right lines. It meant I was able to get rid of a lot of places where we were looking up things using serialized ids in a `static` way, which made the code harder to test (because we couldn't subsitute some alternative lookup). I have yet to actually write tests which use these abstractions, but you might want to take a look and let me know whether it makes sense to you.